src/main/java/net/atos/zac/app/zaken/ZakenRESTService.java

```diff
@@ -1006,22 +1019,55 @@ private int verlengOpenTaken(final UUID zaakUUID, final int duurDagen) {
-        zrcClientService.patchZaak(deelzaakUUID, deelzaakPatch);
-        System.out.println("koppelHoofdEnDeelzaak ZAAK.updated " + hoofdzaak.getUuid());
+    private void koppelHoofdEnDeelzaak(final Zaak hoofdZaak, final Zaak deelZaak) {
+        final HoofdzaakZaakPatch zaakPatch = new HoofdzaakZaakPatch(hoofdZaak.getUrl());
+        zrcClientService.patchZaak(deelZaak.getUuid(), zaakPatch);
         // Hiervoor wordt door open zaak alleen voor de deelzaak een notificatie verstuurd.
         // Dus zelf het ScreenEvent versturen voor de hoofdzaak!
-        indexeerService.addZaak(hoofdzaak.getUuid(), false);
-        eventingService.send(ZAAK.updated(hoofdzaak.getUuid()));
+        indexeerService.addZaak(hoofdZaak.getUuid(), false);
+        eventingService.send(ZAAK.updated(hoofdZaak.getUuid()));
     }
 
-    private void ontkoppelHoofdEnDeelzaak(final UUID deelzaakUUID, final UUID hoofdzaakUUID, final String reden) {
-        final HoofdzaakZaakPatch deelzaakPatch = new HoofdzaakZaakPatch(null);
-        zrcClientService.patchZaak(deelzaakUUID, deelzaakPatch, reden);
+    private void ontkoppelHoofdEnDeelzaak(final Zaak hoofdZaak, final Zaak deelZaak, final String reden) {
+        final HoofdzaakZaakPatch zaakPatch = new HoofdzaakZaakPatch(null);
+        zrcClientService.patchZaak(deelZaak.getUuid(), zaakPatch, reden);
         // Hiervoor wordt door open zaak alleen voor de deelzaak een notificatie verstuurd.
         // Dus zelf het ScreenEvent versturen voor de hoofdzaak!
-        indexeerService.addZaak(hoofdzaakUUID, false);
-        eventingService.send(ZAAK.updated(hoofdzaakUUID));
+        indexeerService.addZaak(hoofdZaak.getUuid(), false);
+        eventingService.send(ZAAK.updated(hoofdZaak.getUuid()));
+    }
+
+    private void koppelRelevantezaken(final Zaak zaak, final Zaak andereZaak, final AardRelatie aardRelatie) {
+        final RelevantezaakZaakPatch zaakPatch = new RelevantezaakZaakPatch(
+                addRelevanteZaak(zaak.getRelevanteAndereZaken(), andereZaak.getUrl(), aardRelatie));
+        zrcClientService.patchZaak(zaak.getUuid(), zaakPatch);
+    }
+
+    private void ontkoppelRelevantezaken(final Zaak zaak, final Zaak andereZaak, final AardRelatie aardRelatie,
+            final String reden) {
+        final RelevantezaakZaakPatch zaakPatch = new RelevantezaakZaakPatch(
+                removeRelevanteZaak(zaak.getRelevanteAndereZaken(), andereZaak.getUrl(), aardRelatie));
+        zrcClientService.patchZaak(zaak.getUuid(), zaakPatch, reden);
+    }
+
+    private List<RelevanteZaak> addRelevanteZaak(final List<RelevanteZaak> relevanteZaken, URI andereZaak,
+            final AardRelatie aardRelatie) {
+        final RelevanteZaak relevanteZaak = new RelevanteZaak(andereZaak, aardRelatie);
+        if (relevanteZaken != null) {
+            relevanteZaken.add(relevanteZaak);
+            return relevanteZaken;
+        }
+        return List.of(relevanteZaak);
```

Comment:
Ik werd hier even op het verkeerde been gezet. Zet even een else voor deze regel. Ik zag de return niet in de if.

Reply:
Pas ik aan.